src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +60,32 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        // Refresh connection is best effort. it should never throw exception
+        private async Task RefreshConnectionAsync(Uri jobServerUrl, VssCredentials jobServerCredential, DelegatingHandler[] delegatingHandler)
+        {
+            Trace.Info($"Refresh JobServer VssConnection to get on a different AFD node.");
+
+            VssConnection newConnection = null;
+
+            try
+            {
+                _hasConnection = false;
+                newConnection = VssUtil.CreateConnection(jobServerUrl, jobServerCredential, delegatingHandler);
+                await newConnection.ConnectAsync();
+                _connection = newConnection;
+            }
+            catch (Exception ex)
+            {
+                Trace.Error($"Catch exception during resetting JobServer connection.");
+                Trace.Error(ex);
+                newConnection?.Dispose();
+            }
+            finally
+            {
+                _hasConnection = true;
```

Comment:
i think we can remove this as well.

Reply:
I ended up removing the other extra `_hasConnection`, to keep the `true` and `false` pair together in one method.